src/hotspot/share/opto/node.cpp

```diff
@@ -1644,78 +1636,72 @@ Node* find_node_by_idx(Node* start, uint idx, bool traverse_output, bool only_ct
   return result;
 }
 
-int node_idx_cmp(Node** n1, Node** n2) {
+int node_idx_cmp(const Node** n1, const Node** n2) {
   return (*n1)->_idx - (*n2)->_idx;
 }
 
-Node* find_node_by_name(Node* start, const char* name) {
+void find_node_by_name(Node* start, const char* name) {
   ResourceMark rm;
-  Node* result = nullptr;
-  GrowableArray<Node*> ns;
-  auto callback = [&] (Node* n) {
+  GrowableArray<const Node*> ns;
+  auto callback = [&] (const Node* n) {
     if (StringUtils::is_star_match(name, n->Name())) {
       ns.push(n);
-      result = n;
     }
   };
   visit_nodes(start, callback, true, false);
   ns.sort(node_idx_cmp);
   for (int i = 0; i < ns.length(); i++) {
     ns.at(i)->dump();
   }
-  return result;
 }
 
-Node* find_node_by_dump(Node* start, const char* pattern) {
+void find_node_by_dump(Node* start, const char* pattern) {
```

Comment:
Since we now only dump nodes, how about renaming this method to `dump_nodes_by_dump()` and only keep `find_node(s?)_by_dump()` to call it from the debugger? Same for the other changed `find*()` methods.

Reply:
I did think about renaming it do `dump_...`. But then I also find it important that the name says that we do filter / search / find.

Reply:
The filter/search action is probably implied but I don't have a strong opinion about it -  it's fine to leave the name like that. But I suggest to make it plural (`find_nodes_by_dump()`) as we are possibly returning multiple nodes.

Reply:
I like the idea with the plural. But I will keep the `find`